File: src/UserProfile.cpp
#include "UserProfile.h"

UserProfile::UserProfile(std::string id, int a) : patientId(id), age(a) {}

UserProfile::UserProfile() : patientId("None"), age(0) {}

void UserProfile::addHealthIndicator(const std::string& indicator, bool value) {
    healthIndicators[indicator] = value;
}

File: src/UserProfile.java
package src;

import java.util.HashMap;
import java.util.Map;

public class UserProfile {
    public String patientId;
    public int age;
    public Map<String, Boolean> healthIndicators;

    public UserProfile(String patientId, int age) {
        this.patientId = patientId;
        this.age = age;
        this.healthIndicators = new HashMap<>();
    }

    public void addHealthIndicator(String key, boolean value) {
        this.healthIndicators.put(key, value);
    }
} 